// contlo-web-sdk/src/utils.js
export function addContloScript(apiKey) {
    const scriptTag = document.createElement('script');
    scriptTag.type = 'text/javascript';
    scriptTag.async = true;
    scriptTag.src = `https://marketing.contlo.com/js/contlo_messaging_v3.js?v=3&shop_id=${apiKey}`;

    const head = document.head || document.getElementsByTagName('head')[0];
    head.appendChild(scriptTag);
}

export async function urlValidator(urls) {
    const urlPattern = /^(https?|ftp):\/\/[^\s/$.?#].[^\s]*$/i;

    if (!urlPattern.test(urls)) {
        console.error('Enter Valid Url');
        return false;
    }

    try {
        const response = await fetch('https://events.contlo.com/update-urls', {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json',
            },
            body: JSON.stringify({ urls }),
        });

        if (response.ok) {
            const data = await response.json();
            console.log('Success:', data);
            return true;
        } else {
            console.error('Error:', response.status);
            return false;
        }
    } catch (error) {
        console.error('Error:', error);
        return false;
    }
}


export async function apiKeyValidator(apiKey) {
    try {
        const response = await fetch('https://marketing.contlo.com/v1/shop_api_details', {
            method: 'GET',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json',
                'X-Api-Key': apiKey
            }
        });

        if (response.ok) {
            const data = await response.json();
            console.log('Success:', data);
            return true;
        } else {
            console.error('Error:', response.status);
            return false;
        }
    } catch (error) {
        console.error('Error:', error);
        return false;
    }
}
